packages/react/src/grid/GridCell.tsx

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright (c) 2023 Gemeente Amsterdam
+ */
+
+import clsx from 'clsx'
+import { PropsWithChildren } from 'react'
+import { PageGridColumnNumber } from './PageGrid'
+
+export type GridCellProps = PropsWithChildren<{
+  gridColumns?: [PageGridColumnNumber, PageGridColumnNumber]
```

Comment:
This is a compact API. Is it readable enough? Can people infer what it means and get used to it?

The obvious alternative is `<GridCell start={1} end={4}>`. Probably even better. But to allow all components to be placed on a grid, these prop names are too generic. I’d say we’d prefer something like `<Component grid-column-start={1} grid-column-end={4}>` or `<Component grid-columns={[1, 4]}>`.

(If we decide to keep a grid cell component, we could use unprefixed props for that component only: `<GridCell columns={[1, 4]}>`.

Additionally, we must consider the need to specify grid placement for all breakpoints, of which we currently have two, e.g. `<Component grid-columns={{narrow: [1, 1], wide: [1, 4]}>`. Is this acceptable?

I’ve kept `grid-span` out of scope for now, and named grid cells as well. We might use the latter for page templates components.

Reply:
Hmm, looking at the example with paragraphs, using spans could offer more simplicity, e.g.

```
  <Paragraph grid-columns={3}>…</Paragraph>
  <Paragraph grid-columns={9}>…</Paragraph>
```

The user would then be responsible to find numbers correctly adding up to 12. And for more creative layouts, we would need to combine props, e.g.:

```
  <Paragraph grid-column-start={2} grid-columns={3}>…</Paragraph>
  <Paragraph grid-columns={7}>…</Paragraph>
```

Reply:
> `gridColumns={[1, 4]}`

I'm not sure about this API, it doesn't feel very intuitive to me. Wouldn't using an object there work?

Something like:

`gridColumns={{ start: 1, end: 4 }}`

And for above and below breakpoint:

`gridColumns={{ narrow: { start: 1, end: 4 }, wide: { start: 1, end: 8 }}}`

I think Typescript will be more of a help with objects as well. 

I'd personally opt for the `span` approach over the `end` approach, it's easier for me to say 'start at column 3 and span 6 columns' than 'start at column 3 and then stop at 3 plus 6 is 9, so column 9'.

Reply:
Yes, touched on the multiple-breakpoint API above, moved out of scope for this PR.

The object notation is a bit verbose but much clearer, certainly after supporting multiple breakpoints.

I agree with starting with `start` and `span` rather than `start` and `end`.

Reply:
I even made `start` optional and a single `span` the default, so we can abbreviate `<Component gridColumns={{start: 1, span: 4}}>` to `<Component gridColumns={4}>`.